src/test/java/com/snowplowanalytics/snowplow/tracker/TrackerTest.java

```diff
@@ -111,6 +110,8 @@ public void testEcommerceEvent() {
                 .build());
 
         // Then
+        Thread.sleep(500);
```

Comment:
I'm surprised we have to slow the tests down this much, half a second is quite the delay, especially when we're dealing with a mock emitter. This makes me think of the F in [FIRST](https://dzone.com/articles/writing-your-first-unit-tests).

As an aside, this also makes me curious about the performance of this change, does is have similar perfromance properties to previous releases? A new threading paradigm could have some unexpected consequences that we'd be wise to test more deeply with some performance testing.

Reply:
I didn't try shorter waits - 500ms was already being used for the Emitter tests. I'll see if it can be improved.

Reply:
I tried shorter wait times, but they all failed tests occasionally, even 450ms.

Reply:
Does this suggest that even in a "real" environment, the difference between created timestamp and sent timestamp is at least 500ms? That seems reasonable high to me. Might be out of scope here but worth investigating the cause if true.

Reply:
Have we got any "real" Java tracker data lying around anywhere which we could compare timestamps?

Reply:
This is out of scope here yeah. We're planning to finish this PR soon without looking into performance. The simple-console is broken currently, so I'll fix that next and then extend it for performance testing.

Reply:
I don't think the test is a good parameter to compare the performance. The Thread.sleep is mostly needed to allow the Thread to start and do the job before the test is finished and it mostly depends by the machine where they are running on. I think, if we have concerns on this we should compare the two implementations in a real test with simple-console as suggested. However, we need to decide what we want to test. I guess we want to test how fast the track method can return back. You mentioned the difference between created_tstamp and sent_tstamp but even if the gap is longer the derived_tstamp would mitigate the discrepancy. We base our data modelling on the derived_tstamp, so even a longer gap between the two shouldn't be disruptive for the customer. IMO if we are really concerned about the performance of the `track` method we could add a buffer in the input consumed by the tracker thread-pool which would make the performance exactly the same as before.